### ISSUE TITLE
Support custom NavbarBrand instead of just a string?

### ISSUE THREAD

**Issue by mkhorton** [ASIDataScience/dash-bootstrap-components#101]
Hi, I'm not sure if I'm missing an obvious solution to this, but it seems like as currently written the Navbar brand has to be a string, even though the NavbarBrand component itself could in principle contain other items as children (eg an Html.Img)?

I'm looking for an easy way to add a left-aligned logo to a Navbar instead of only brand text.

Many thanks for releasing this component suite, it looks very useful.

**Comment by mkhorton** [comment]
Actually may be easier to do this via css and the brand_style.

**Comment by tcbegley** [comment]
Hey @mkhorton, you're right, the brand does have to be a string. 

Currently it's only possible to pass Dash components as the children of other Dash components, not any other prop. So in particular you can't pass, e.g. `html.Img` to `brand`. There was [an issue](https://github.com/plotly/dash-renderer/pull/32) opened a while back on the `dash-renderer` repo that would have addressed this, but it's a big change and has been dormant for a little while, so I don't expect it to happen any time soon.

This restriction, and also because we were trying to keep the interface of `Navbar` relatively simple, meant that we sacrificed some customisability of the `Navbar` component.

I posted on the [plotly community forums](https://community.plot.ly/t/show-and-tell-dash-bootstrap-components/16614/15?u=tcbegley) about how you could more manually construct a Bootstrap style navbar using `dash-html-components` and applying the relevant Bootstrap classes with `className`. Perhaps that's of use to you for now.

As a longer-term fix, since multiple people are asking similar questions, perhaps we can create a `CustomNavbar` component that gives you full control over the contents, but would require some careful construction of the `children`. @pbugnion what do you think?

**Comment by pbugnion** [comment]
I agree with the idea of a new custom component that gives more control over the children. It seems like a good way to give the user (almost) arbitrary flexibility, but still keeping a simple navbar for people who don't need all that flexibility.

**Comment by ned2** [comment]
I would love a custom component that supports more control over the children. I just came here to open an issue asking about just this.

In addition to the limitation of only having strings as the `brand` prop, I was also hoping to be able to add Dash components to parts of the Navbar that are not nav-items. For example, including an form element before the nav items, as in [this Bootstrap dashboard](https://getbootstrap.com/docs/4.1/examples/dashboard).

**Comment by tcbegley** [comment]
This has been resolved by #118 which has been released as version `0.3.0` of *dash-bootstrap-components*. You now have full control over the children of `Navbar`. We've also added a `NavbarSimple` component that has the same functionality as the old `Navbar` in case you don't need the flexibility of the new component.

Check out [the docs](https://dash-bootstrap-components.opensource.asidatascience.com/l/components/navbar) or the [Navbar examples](https://github.com/ASIDataScience/dash-bootstrap-components/blob/master/examples/advanced-component-usage/Navbars.py) for more.
